src/hooks/useLocalAndRemotePatients.js

```diff
@@ -99,15 +128,44 @@ export const useLocalAndRemotePatients = (initialValue = []) => {
   }, [fetchError]);
 
   const onPressSearchOnline = searchParams => {
+    const paramsWithLimits = { ...searchParams, limit, offset };
+
     refresh();
+
     fetch(
-      getPatientRequestUrl(searchParams),
+      getPatientRequestUrl(paramsWithLimits),
       { headers: { authorization: getAuthorizationHeader() } },
       { responseHandler: processPatientResponse }
     );
+    dispatch({ type: 'fetch_more' });
+  };
+
+  const getMorePatients = async searchParams => {
+    if (!response) return;
+
+    dispatch({ type: 'getting_more_patients' });
+    const paramsWithLimits = { ...searchParams, limit, offset };
+
+    // Use RNFetch as the fetch returned from `useFetch` is coupled with it's state in a specific
+    // implementation, which we want to change by appending to the result, rather than replacing.
+    const getMoreResponse = await RNFetch(
+      `${getServerURL()}${getPatientRequestUrl(paramsWithLimits)}`,
+      {
+        headers: { authorization: getAuthorizationHeader() },
+      }
+    );
+
+    dispatch({
+      type: 'add_more_patients',
+      payload: {
+        data: processPatientResponse({ ...getMoreResponse, json: await getMoreResponse.json() }),
```

Comment:
`processPatientResponse` throws an error if there are no results.. which happens when we hit the end of the patient list. 
I think it should dispatch an action that keeps the current list of patients and sets `gettingMore: false`